src/main/java/juja/sqlcmd/DatabaseManager.java

```diff
@@ -30,4 +33,83 @@ public boolean connect(String database, String user, String password) {
             return false;
         }
     }
+
+    public void close() throws SQLException {
+        if (connection != null) {
+            connection.close();
+        }
+    }
+
+    public String[] getTableNames() throws SQLException {
+        try (ResultSet rs = connection.getMetaData().getTables(
+                null, "public", "%", new String[]{"TABLE"})) {
+            int tablesCount = 0;
+            if (rs.last()) {
+                tablesCount = rs.getRow();
+                rs.beforeFirst();
+            }
+            String[] tableNames = new String[tablesCount];
+            int index = 0;
+            while (rs.next()) {
+                tableNames[index++] = rs.getString("table_name");
+            }
+            return tableNames;
+        }
+    }
+
+    public DataSet[] getTableData(String tableName) throws SQLException {
+        if (!isTableExists(tableName)) {
+            return new DataSet[0];
+        }
+        int tableSize = TableSize(tableName);
+        if (tableSize == 0) {
+            return new DataSet[0];
+        }
+        DataSet[] dataSets = new DataSet[tableSize];
+        int numberOfColumns = NumberOfColumns(tableName);
+        fillDataSets(tableName, dataSets, numberOfColumns);
```

Comment:
Explain why you doing twice select.
I think that once will be enough.
What do you think about this:
```
private void fillDataSets(String tableName, DataSet[] result) throws SQLException {
         try (Statement stmt = connection.createStatement();
              ResultSet rs = stmt.executeQuery("SELECT * FROM public." + tableName)) {
             int index = 0;
             // This line instead the method NumberOfColumns(String tableName)
             int numberOfColumns = rs.getMetaData().getColumnCount(); 
             while (rs.next()) {
                 DataSet dataSet = new DataSet(numberOfColumns);
                 result[index++] = dataSet;
                 for (int i = 0; i < numberOfColumns; i++) {
                     dataSet.insertValue(i, rs.getString(i + 1));
                 }
             }
         }
     }
            
```
I think the method private int NumberOfColumns(String tableName) we can delete.

Reply:
I need numberOfColumns in the getTableData method, so I extracted it. Yes I saw that I select db twice but it's a compromise. Also I can reuse this method with other tables